src/actions/charts.ts

```diff
@@ -43,25 +46,38 @@ export function fetchCharts(repo: string) {
   };
 }
 
-export function getChart(id: string) {
+export function fetchChartVersions(id: string) {
   return (dispatch: Dispatch<IStoreState>): Promise<{}> => {
     dispatch(requestCharts());
-    return fetch(url.api.charts.get(id))
+    return fetch(url.api.charts.listVersions(id))
       .then(response => response.json())
-      .then(json => {
-        const c: IChart = json.data;
-        dispatch(listChartVersions(c.id));
-        dispatch(getChartReadme(c.id, c.relationships.latestChartVersion.data.version));
-        return dispatch(selectChart(json.data));
-      });
+      .then(json => dispatch(receiveChartVersions(json.data)));
   };
 }
 
-export function listChartVersions(id: string) {
+export function fetchChartVersionsAndSelectVersion(id: string, version?: string) {
   return (dispatch: Dispatch<IStoreState>): Promise<{}> => {
-    return fetch(url.api.charts.listVersions(id))
-      .then(response => response.json())
-      .then(json => dispatch(receiveChartVersions(json.data)));
+    return dispatch(fetchChartVersions(id)).then((action: any) => {
+      const versions: IChartVersion[] = action.versions;
+      let cv: IChartVersion = versions[0];
+      if (version) {
+        const found = versions.find(v => v.attributes.version === version);
+        if (!found) {
+          throw new Error("could not find chart version");
+        }
+        cv = found;
+      }
+      dispatch(getChartReadme(id, cv.attributes.version));
+      return dispatch(selectChartVersion(cv));
+    });
+  };
+}
+
+export function selectChartVersionAndGetReadme(cv: IChartVersion) {
```

Comment:
I don't like the idea of chaining actions in the react components so I decided to create new actions like this that chain other actions. I think it's better for the actions to handle the flow of different requests and for the components to just call one simple method. WDYT of this approach @evanlouie?